File: leetcode/src/evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        vector<int> stack;
        for (auto it = tokens.begin(); it != tokens.end(); it ++) {
            string token = *it;

            if (token == "+") {
                int op2 = stack.back();stack.pop_back();
                int op1 = stack.back();stack.pop_back();
                stack.push_back(op1 + op2);
            } else if (token == "-") {
                int op2 = stack.back();stack.pop_back();
                int op1 = stack.back();stack.pop_back();
                stack.push_back(op1 - op2);
            } else if (token == "*") {
                int op2 = stack.back();stack.pop_back();
                int op1 = stack.back();stack.pop_back();
                stack.push_back(op1 * op2);
            } else if (token == "/") {
                int op2 = stack.back();stack.pop_back();
                int op1 = stack.back();stack.pop_back();
                stack.push_back(op1/op2);
            } else {
                stack.push_back(stoi(token));
            }
        }

        return stack.back();
    }
};

File: leetcode/src/evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []

        for token in tokens:
            if (token.isdigit() or (token != "-" and token.lstrip("-").isdigit())):
                stack.append(int(token))
                continue
            
            op2 = stack.pop()
            op1 = stack.pop()

            if (token == "+"):
                stack.append(op1 + op2)
            elif (token == "-"):
                stack.append(op1 - op2)
            elif (token == "*"):
                stack.append(op1 * op2)
            elif (token == "/"):
                stack.append(int(op1/op2))
            
        
        return stack.pop()